### DISCUSSION TITLE
Print line breaks in in scopes

### DISCUSSION THREAD

**Discussion by ubaumann** [Textualize/rich#2974] [Q&A] [answered]
What is the best way to print newlines in a string using `render_scope`?

I have the following code example:

```python
>>> from rich.scope import render_scope
>>> from rich import print
>>>
>>> long_text = """This is
... a text with
... multiple
... line breaks"""
>>> short_text = "Rich is AWESOME"
>>> data = {"long_text": long_text, "shorty": short_text}
>>>
>>> print(render_scope(data))
╭───────────────────────────────────────────────────────────╮
│ long_text = 'This is\na text with\nmultiple\nline breaks' │
│    shorty = 'Rich is AWESOME'                             │
╰───────────────────────────────────────────────────────────╯
>>>
```

I would like to be able to tell Rich to print the line breaks in the string. So the output would look something like this:

```
╭───────────────────────────────────────────────────────────╮
│ long_text = 'This is                                      │
│              a text with                                  │
│              multiple                                     │
│              line breaks'                                 │
│    shorty = 'Rich is AWESOME'                             │
╰───────────────────────────────────────────────────────────╯
```

**Comment by willmcgugan** [comment, marked as answer]
`render_scope` is intended to display something like a `repr`, so it will always escape newlines as Python does.

Only way I can think of to change it would be to copy `render_scope` and add a special case for strings.

**Reply by ubaumann** [reply]
Thank you so much for your fast answer!!
I am using the render_scope in the print functionality shown in the screenshot here: https://github.com/InfrastructureAsCode-ch/nornir_rich#print-result

Maybe render_scope is not the best solution and I should use a table or so.

I will try to avoid copying the code and changing it for my use case out of maintainability but it is good to know I did not miss something obvious.
